server/models/handlers/account.js

```diff
@@ -1,24 +1,43 @@
 /* The goal: Take in the user input from the Account Creation form in 
 the front end UI in order to create a user account in the backend*/
 
-async function createUserAccount(firstName, lastName, type, studentID, email, password, phone) {
-    
-    const user = {
-        type: 'student', //note: 'type' is called 'role' in the UI, but refers to either a worker or a student.
-        studentId: '20764242',
-        firstname: 'Victor',
-        lastname: 'Hugo',
-        email: 'victorhugo@gmail.com',
-        phone: '6476442200'
-
-         /*  Please note: When returning user object, 
-        I have omitted the 'password'. 
-        This is done intentionally for privacy reasons. */
-    };
+async function createUserAccount(firstName, lastName, type, studentId, email, password, phone) {
+    // Searches the student table to see if a student account exists for the given data.
+var user;
+    If userType = userType.student {
+        //Generates an accessToken of length 20 characters.
+        // Note: The accessToken includes all alphanumeric characters except for 0, O, I, and l — characters easily mistaken for each other)
+       const accessToken =  new UIDGenerator(UIDGenerator.BASE16).20; 
 
+        user = await studentModel.insertStudentAccount(firstName, lastName, type, studentId, email, password, phone);
+        if (!_.isNil(student)) return student;
```

Comment:
Hi Nancy,
I would suggest adjusting this statement to check if the `user` variable is nil rather than `student` as the student variable does not exist.

Reply:
Hi Amy,
That makes sense; I have adjusted the code to check the  if user object is Nil, rather than the student object. You can see the change in line 13 of server/models/handlers/account.js